### DISCUSSION TITLE
Experimental UI navigation

### DISCUSSION THREAD

**Discussion by braunjj** [dagster-io/dagster#21370] [General]
We’re testing some changes to the top nav bar in the app UI to make it easier to explore and navigate the UI. This experiment applies to both Dagster Core and Dagster+ and you can enable it from your user settings in the upcoming release (1.7.3).

### What are we aiming to solve:
**The left sidebar doesn't scale well:**
- Some organizations have hundreds (or thousands) of asset groups and job and we want to provide better ways to organize these definitions. We recently introduced Owners and Tags for assets and have plans to add tags to other definition types as well but we are running out of space in the sidebar to add these new organization and filtering tools.

**Too many sidebars:**
- We've added sidebars to many other views in the product recently which often feel odd alongside the app-wide sidebar. Some views of the app have two or three sidebars which isn't great.

**"Overview" is overloaded:**
- The existing Overview tab is a grab bag of status pages and definition indexes. We've heard feedback that users sometimes feel like they need to go hunting around the UI to find certain pages (especially new users). Pages like Schedules, Sensors or Backfills don't feel natural under "Overview".

----------

### Changes we're exploring
- Removing the sidebar and giving the Assets and Jobs top level index pages – this gives us more room to introduce improved grouping and filtering tools (like we recently did for assets)
> Update: We're going to hold off on removing the sidebar for now
- Creating a new Automation tab where Sensors, Schedules will live
----------

This is the first navigation experiment we are running. There are still some oddities that I'd like to clean up and you can expect us to iterate on this a few times over the coming months. For example, the "Resources" page still feels randomly placed under "Overview".

We'd appreciate any feedback or input you have. You can share it right here. Our ultimate goal is to make the UI easier to navigate and explore (especially for new users).

**Comment by nils-borrmann-tacto** [comment]
Thanks for improving this, definitely a step in the right direction! 

One thing that still feels odd to me: When I'm on the "Jobs" page and then I click on one job, suddenly I'm in the "Settings" part of Dagster. That just doesn't seem right in a way.

**Reply by braunjj** [reply]
Thanks @nils-borrmann-tacto. Appreciate the feedback. I agree. I think this is just a bug due to the way we feature flagged this. We'll get this cleaned up.

**Comment by axellpadilla** [comment]
Excelent work! (using 1.7.4 because dependencies with dbt-sqlserver.)

I though it was deployment because literally you need to clic reload to reload code changes and definitions, but settings and grouping it there is very good, but I hope there is a more automatic way to make it for example with a git hook

**Comment by ClaytonSmith** [comment]
![image](https://github.com/dagster-io/dagster/assets/8992035/f7584908-b17a-4dc4-a747-1a533ff84f9e)

I will literally do whatever it takes to get these green dots to be yellow when the partition is outdated. I cannot understate how much time has been spent searching for outdated partitions. 

Please. Please, please, please, please, please, please, please, please, please, please, please, please, please, please, please, please, x10000. Please this is so valuable

**Comment by zyd14** [comment]
Please don't take away the jobs sidebar. Maybe collapse it by default, or make it a toggle, but please don't take it away. I find it very handy to be able to go straight to a job's launchpad after checking the overview, without having to open up the jobs page

**Comment by itaisoares** [comment]
for me when I'm at the overview tab, I expect to see what did wrong or well and if it didn't run ok I would like to run again easily (or go to the details to see what happened, but this part is ok), that means that going to the job page to start it again is not a good flow for our scenario, we have like 15-16 deployments and sometimes 4-5 jobs fails so I need to enter each job and start again or open new tabs for each job that I want to restart.

**Comment by sheffardj** [comment]
Definitely a great and appreciated change! Can you separate out sensors from schedules a bit better on the automation page? Mixing those together has not been great for us. Being able to quickly view the status of all our sensors was a blessing, now we're having to search for each one among hundreds of schedules and doesn't give us a good sense of the heartbeat of our pipelines. In the Overview page you have three tabs (Timeline, Resources, Backfills), something similar for Automation (Schedules, Sensors) would be awesome!

**Reply by axellpadilla** [reply]
Hi @sheffardj  just wanted to check if you know that on filters there is a filter to see only schedules and only sensors:
![image](https://github.com/user-attachments/assets/935b4494-6cca-47bb-a367-48297f216b01)

**Comment by clayheaton** [comment]
It's nice to be able to collapse the sidebar. However, I would prefer an option to leave it locked in place. Apache Superset handles this well with the Dashboard filters sidebar. Here it is expanded:

![Screenshot 2024-10-18 at 10 46 28 AM](https://github.com/user-attachments/assets/9336a9f2-eaf8-4439-9588-5c5b71ce35d9)

and here it is collapsed:

![Screenshot 2024-10-18 at 10 47 27 AM](https://github.com/user-attachments/assets/74c045fc-a6c1-4326-b19a-7dab3fa5282b)

**Comment by alexanderfifefd** [comment]
Older versions of Dagster used to "lock" the left nav bar. Newer versions seem to always collapse the sidebar when navigating. Or I may be missing a setting or something.

I use the jobs in the left nav bar heavily. Close to 80% of the time I use Dagster I'm either in a job view or a run/asset view. Most of the time the next place I go to is to another Job which I then use to navigate to the specific assets I want. I find the graph view very helpful for quickly navigating to new components and prefer it over the list views you find in "Overview" and "Runs".

Currently, as the UI stands, I either have to click the Hamburger menu an extra time every time I want to navigate to a job. Or click on "Overview" -> "Job" -> "Specific Job" and then go to the previous page to go to a new asset. 

Essentially you've added an extra click to my most used navigation patterns. Please add a setting for "lock the sidebar" or a small toggle like Clay showed from Superset. 

Seems like a minor thing, but it's pretty unergonomic for me at the moment.
